Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -1139,6 +1140,9 @@ private extension PurchasesOrchestrator {
                         return
                     }
 
+                    let appTransactionJWS = await self.transactionFetcher.appTransactionJWS(
+                        refreshPolicy: refreshPolicy)
```

Comment:
Moved fetching the `AppTransaction` to after the check to see if we have transactions, originalPurchaseDate and originalApplicationVersion.

This way, if we don't need to post the receipt, we wouldn't be fetching the AppTransaction unnecessarily, potentially showing the authentication prompt.